bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -139,24 +151,33 @@ protected void activate(Map<String, Object> properties) {
             this.itemRegistry.addRegistryChangeListener(itemRegistryListener);
             this.thingRegistry.addRegistryChangeListener(thingRegistryListener);
         } else {
+            logger.debug("Disabling channel item provider.");
             for (ProviderChangeListener<Item> listener : listeners) {
                 for (Item item : getAll()) {
                     listener.removed(this, item);
                 }
             }
+            removeRegistryChangeListeners();
         }
     }
 
     protected void deactivate() {
-        this.itemRegistry.removeRegistryChangeListener(itemRegistryListener);
-        this.linkRegistry.removeRegistryChangeListener(linkRegistryListener);
-        this.thingRegistry.removeRegistryChangeListener(thingRegistryListener);
+        removeRegistryChangeListeners();
         synchronized (this) {
             items = null;
         }
     }
 
+    private void removeRegistryChangeListeners() {
```

Comment:
Wouldn't it be more consistent if we also create a `addRehistryChangeListeners()` method and call it [here](https://github.com/eclipse/smarthome/pull/1824/files#diff-6e13cc414ff80d2180afe02e0907f1e8R150)?

So the function could be near to the other and it is easier to identify that changing the one needs to change the other, too.